src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs

```diff
@@ -138,8 +138,7 @@ public async Task Invoke(HttpContext httpContext)
 
         internal async Task<bool> TryServeCachedResponseAsync(ResponseCachingContext context, IResponseCacheEntry cacheEntry)
         {
-            var cachedResponse = cacheEntry as CachedResponse;
```

Comment:
Filed https://github.com/dotnet/docs/issues/9834 since this is different from what the docs say but it does seem to work.